lib/client.js

```diff
@@ -285,14 +321,7 @@ extend(Raven.prototype, {
     // todo: better property name than sentryContext, maybe __raven__ or sth?
     wrapDomain.sentryContext = options;
 
-    var self = this;
-    if (typeof onErr !== 'function') {
-      onErr = function (err) {
-        self.captureException(err);
-      };
-    }
-
-    wrapDomain.on('error', onErr);
+    wrapDomain.on('error', this.makeErrorHandler());
```

Comment:
meant to encapsulate the `firstError` and `calledWhatever` flags; we have a separate fatal error catcher/handler for each domain and then one for global uncaught exceptions. since domains are sort of heavy anyway, I don't consider it a big cost. it miiight be possible to combine all the domain ones to one set of flags, but I feel like there'd be some potential resulting contrived probably-never-happen-but-might-mess-us-up-if-it-ever-did cases that I'd rather not go thinking about if we can avoid it for relatively little cost.